src/fsm/parser.act

```diff
@@ -44,23 +44,32 @@
 	typedef char *      string;
 	typedef fsm_state_t state;
 
+	struct act_statelist {
+		char *id;
+		fsm_state_t state;
+		struct act_statelist *next;
+	};
```

Comment:
I wonder if we shouldn't get rid of the idea of a string `id` for the fsm(1) syntax entirely. But maybe not, writing `1a`, `1b` etc is convenient for some tests, especially when the expected output is combined from two FSMs.

Reply:
I'd prefer to keep that flexible, it's been helpful when generating test data too.